### PR TITLE
[bug-fix] Use correct memories for LSTM SAC

### PR DESCRIPTION
**Pull request by ervteng**
### Proposed change(s)

Use the right memories for SAC. Now that we are padding in the right direction, LSTM is much more sensitive to getting the memories right. Since SAC needs both the memory and the next memory (for the target network), we have to evaluate the next memory to get it. 

Pending experiment results.

### Types of change(s)

- [x] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/main/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/main/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/main/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: ml-agents/mlagents/trainers/sac/optimizer_torch.py [hunk 1]
```diff
@@ -568,6 +555,16 @@ def update(self, batch: AgentBuffer, num_sequences: int) -> Dict[str, float]:
             q1_stream, q2_stream = q1_out, q2_out
 
         with torch.no_grad():
+            # Since we didn't record the next value memories, get them.
+            if value_memories is not None:
+                just_first_obs = [
+                    _obs[:: self.policy.sequence_length] for _obs in current_obs
+                ]
+                _, next_value_memories = self._critic.critic_pass(
```
Comment (andrewcoh) [inline ml-agents/mlagents/trainers/sac/optimizer_torch.py:563]: should this be generated with `next_obs`? don't understand `just_first_obs`

Comment (andrewcoh) [reply ml-agents/mlagents/trainers/sac/optimizer_torch.py:563]: never mind I think I understand now.

**Review by andrewcoh — APPROVED** [review]
